pkg/datastore/database.go

```diff
@@ -140,6 +140,9 @@ func (db *relationalDb) GetTransaction(ctx context.Context, records ...Record) (
 		if err != nil {
 			return nil, err
 		}
+		if IsMultiInstanced(record, tableName, db.instancer != nil) {
+			tenancyInfo.DbRole = tenancyInfo.DbRole.GetRoleWithInstancer()
+		}
```

Comment:
Here, you're stating that if multi-tenancy is supported in some entity, the user is gonna be able to read/read & write only specific instance's data. What if someone wants to read/read & write all instances' data?

Reply:
dbInstancer configuration is not per table, hence if dbInstancer is configure service is expected to be single instance based.

Reply:
dbInstancer should not be configured if we don't want to access records by instanceIds